### DISCUSSION TITLE
Problems exporting to JPEG

### DISCUSSION THREAD

**Discussion by Keanottesen** [xyflow/xyflow#1908] [General]
Hi

I have had a solution for some time now, where I export the flow as either an image or pdf

```js 
  const download = (_image, type) => {
      console.log(reactFlowContainer?.current);
      if (type === "image") {
        const a = document.createElement('a');
        a.href = _image;
  
        a.download = createFileName('jpg', label);
        a.click();
      } else {
        const doc = new jsPDF({ orientation: 'l' });
  
        const imgProps = doc.getImageProperties(_image);
  
        const pdfWidth = doc.internal.pageSize.getWidth();
        const pdfHeight = (imgProps.height * pdfWidth) / imgProps.width;
  
        doc.addImage(_image, 'JPEG', 0, 0, pdfWidth, pdfHeight);
        doc.save(`${label}.pdf`);
      }
    };
  
    const handleImage = (type) => takeScreenShot(reactFlowContainer?.current).then(img => download(img, type));
```

This has for some reason stopped working in chrome. Anybody who has had the same problem and found a solution?

**Comment by leejustin** [comment]
It will help if you share what kind of error you are getting as well as your version of the package

**Reply by Keanottesen** [reply]
I am using: 

"react-flow-renderer": "^9.7.0",
"use-react-screenshot": "^1.0.2",

chrome: Version 97.0.4692.71 (Official Build) (arm64)

The error is basically that the screenshot is 1 px width and if I try to hardcode a width for the image it is just white. It is as if the canvas thinks that my react-flow div is empty, however when console logging the element and inspecting it in the moment of the screenshot it seems fine to me.

After some experimenting I have found out that by using html2canvas directly an setting foreignObjectRendering option to true (in chrome) the image will render in the correct aspects. However, here there seems to be an issue with the image in terms of typography. see the attached image.
![35633766 (50)](https://user-images.githubusercontent.com/47158972/154131720-493a0106-00b7-432f-8f89-92029f630e89.jpg)
<img width="1655" alt="Screenshot 2022-02-15 at 20 09 08" src="https://user-images.githubusercontent.com/47158972/154131805-848eab93-2c70-48c3-ab5c-cbfb62361ecf.png">

**Comment by moklick** [comment]
I've created a ["Download Image" example](https://reactflow.dev/docs/examples/misc/download-image/) that uses [html-to-image](https://github.com/bubkoo/html-to-image).

**Reply by happysamyak** [reply]
@moklick given example allows me to take screenshot of flow which is on the screen, not whole if I move flow left or right.

**Reply by moklick** [reply]
it uses the current viewport. If you want a different behaviour, you can mirror your flow, adjust the viewport and take a screenshot from that hidden mirrored flow.
